controllers/rabbitmqcluster_controller.go

```diff
@@ -145,6 +148,17 @@ func (r *RabbitmqClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 		}
 	}
 
+	sts, err := r.statefulSet(ctx, rabbitmqCluster)
+	// The StatefulSet may not have been created by this point, so ignore Not Found errors
+	if client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, err
+	}
+	if statefulSetNeedsPostUpgrade(sts, rabbitmqCluster) {
```

Comment:
Does this error case also need to be wrapped in an IgnoreNotFound wrapper? It's not quite clear to me.

Reply:
No, and if it's not clear to you that's good feedback. `statefulSetNeedsPostUpgrade` checks if the statefulSet is nil, so `markForPostUpgrade` has no need to ignore NotFound errors. I'll move the if nil check out of that function so this is clearer.